docs/operator-manual/applicationset/Generators-Plugin.md

```diff
@@ -0,0 +1,237 @@
+
+### Add a ConfigMap to configure the access of the plugin
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-plugin
+data:
+  token: $plugin.myplugin.token # Alternatively $<some_K8S_secret>:plugin.myplugin.token
+  baseUrl: http://myplugin.plugin.svc.cluster.local.
+```
+
+* `token`: Pre-shared token used to authenticate HTTP request (points to the right key you created in the `argocd-secret` Secret)
+* `baseUrl`: BaseUrl of the k8s service exposing your plugin in the cluster.
+
+### Store credentials
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: argocd-secret
+  namespace: argocd
+  labels:
+    app.kubernetes.io/name: argocd-secret
+    app.kubernetes.io/part-of: argocd
+type: Opaque
+data:
+  ...
+  # The secret value must be base64 encoded **once** 
+  # this value corresponds to: `printf "strong-password" | base64`
+  plugin.myplugin.token: "c3Ryb25nLXBhc3N3b3Jk"
+  ...
+```
+
+#### Alternative
+
+If you want to store sensitive data in **another** Kubernetes `Secret`, instead of `argocd-secret`, ArgoCD knows how to check the keys under `data` in your Kubernetes `Secret` for a corresponding key whenever a value in a configmap starts with `$`, then your Kubernetes `Secret` name and `:` (colon) followed by the key name.
+
+Syntax: `$<k8s_secret_name>:<a_key_in_that_k8s_secret>`
+
+> NOTE: Secret must have label `app.kubernetes.io/part-of: argocd`
+
+##### Example
+
+`another-secret`:
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: another-secret
+  namespace: argocd
+  labels:
+    app.kubernetes.io/part-of: argocd
+type: Opaque
+data:
+  ...
+  # Store client secret like below.
+  # Ensure the secret is base64 encoded
+  plugin.myplugin.token: <client-secret-base64-encoded>
+  ...
+```
+
+### HTTP server
+
+#### A Simple Python Plugin
+
+You can deploy it either as a sidecar or as a standalone deployment (the latter is recommended).
+
+In the example, the token is stored in a file at this location : `/var/run/argo/token`
+
+```
+string-password
+```
+
+```python
+import json
+from http.server import BaseHTTPRequestHandler, HTTPServer
+
+with open("/var/run/argo/token") as f:
+    token = f.read().strip()
+
+class Plugin(BaseHTTPRequestHandler):
+
+    def args(self):
+        return json.loads(self.rfile.read(int(self.headers.get('Content-Length'))))
+
+    def reply(self, reply):
+        self.send_response(200)
+        self.end_headers()
+        self.wfile.write(json.dumps(reply).encode("UTF-8"))
+
+    def forbidden(self):
+        self.send_response(403)
+        self.end_headers()
+
+    def unsupported(self):
+        self.send_response(404)
+        self.end_headers()
+
+    def do_POST(self):
+        if self.headers.get("Authorization") != "Bearer " + token:
+            self.forbidden()
+        elif self.path == '/api/v1/getparams.execute':
+            args = self.args()
+            self.reply(
```

Comment:
I wonder if we should future proof this response schema. `{"params": []}`

Reply:
If it's to encapsulate the current response in order to potentially add future elements other than "params," then yes, it's a good idea.

Like : 

```json
{
  "params": [
    {
      "digestFront": "sha256:a3f18c17771cc1051b790b453a0217b585723b37f14b413ad7c5b12d4534d411",
      "digestBack": "sha256:4411417d614d5b1b479933b7420079671facd434fd42db196dc1f4cc55ba13ce"
    },
    {
      "digestFront": "sha256:7c20b927946805124f67a0cb8848a8fb1344d16b4d0425d63aaa3f2427c20497",
      "digestBack": "sha256:e55e7e40700bbab9e542aba56c593cb87d680cefdfba3dd2ab9cfcb27ec384c2"
    }
  ],
  "futur": "foo"
}
```